applications/nrf5340_audio/src/modules/hw_codec.c

```diff
@@ -73,59 +75,53 @@ int hw_codec_volume_set(uint8_t set_val)
 	if (ret) {
 		return ret;
 	}
+
+	prev_volume_reg_val = volume_reg_val;
+
+	/* This is rounded down to nearest integer */
+	LOG_DBG("Volume: %" PRId32 " dB", (volume_reg_val / 2) - MAX_VOLUME_DB);
+
 	return 0;
 }
 
 int hw_codec_volume_adjust(int8_t adjustment_db)
 {
 	int ret;
-	static uint32_t prev_volume_reg_val = OUT_VOLUME_DEFAULT;
+	int32_t new_volume_reg_val;
 
 	LOG_DBG("Adj dB in: %d", adjustment_db);
 
 	if (adjustment_db == 0) {
-		ret = cs47l63_write_reg(&cs47l63_driver, CS47L63_OUT1L_VOLUME_1,
-					(prev_volume_reg_val | CS47L63_OUT_VU) &
-						~CS47L63_OUT1L_MUTE);
-		return ret;
-	}
-
-	uint32_t volume_reg_val;
-
-	ret = cs47l63_read_reg(&cs47l63_driver, CS47L63_OUT1L_VOLUME_1, &volume_reg_val);
-	if (ret) {
-		return ret;
-	}
-
-	volume_reg_val &= CS47L63_OUT1L_VOL_MASK;
-
-	/* The adjustment is in dB, 1 bit equals 0.5 dB,
-	 * so multiply by 2 to get increments of 1 dB
-	 */
-	int32_t new_volume_reg_val = volume_reg_val + (adjustment_db * 2);
+		new_volume_reg_val = prev_volume_reg_val;
```

Comment:
Is it necessary to write to HW codec if adjustment is 0?

Reply:
Can return 0 here if not necessary

Reply:
The 0 is used when starting up the chip and will indicate that the volume should be set to the value it had when it was shut down, or the default value when booting up.

Reply:
It's also described in the header file for this function

Reply:
Aha, IMO it would be better to write the initial value directly in hw_codec_default_conf_enable(), but this is good enough